UnityProject/Assets/Scripts/Learning/ProtipObject.cs

```diff
@@ -16,21 +16,31 @@ public class ProtipObject : MonoBehaviour
 		[SerializeField]
 		private bool showEvenAfterDeath = false;
 
-		[SerializeField, HideIf(nameof(triggerOnce)), Tooltip("Will not appear again after a while to not annoy the player with it")]
+		[SerializeField]
 		private float tipCooldown = 200f;
 
 		private bool isOnCooldown = false;
-		private bool hasBeenTriggeredBefore = false;
 
 		private void Awake()
 		{
 			if (TipSO == null)
 			{
 				Logger.LogError("[ProtipObject] - Component missing tip data.");
-				Destroy(this);
+				RemoveThisComponent();
 				return;
 			}
-			if(CheckSaveStatus()) Destroy(this);
+
+			if (CheckSaveStatus()) RemoveThisComponent();
+		}
+
+		/// <summary>
+		/// Ensures that we don't remove any components from the player gameObject.
+		/// </summary>
+		private void RemoveThisComponent()
```

Comment:
I don't get it, from where is it destroyed? Also what issues did the previous flow cause?

Reply:
> where is it destroyed

It's literally written in the comment, we don't want to destroy any protip components while they're on the player.

Reply:
Protips can be applied on basically any prefab
but some protips are on the player object, all the player ones have multiple tips on them so to ensure that the protip doesn't delete itself while the "triggerOnce" variable is on we check if we're on a player before deleting.

Reply:
> Protips can be applied on basically any prefab but some protips are on the player object

this is the piece of context I was missing.